snuba/web/views.py

```diff
@@ -274,6 +274,9 @@ def _trace_transaction(dataset: Dataset) -> None:
             scope.span.set_tag("dataset", get_dataset_name(dataset))
             scope.span.set_tag("referrer", http_request.referrer)
 
+        if scope.transaction:
+            scope.transaction.name = f"{scope.transaction.name}__{get_dataset_name(dataset)}__{http_request.referrer}"
```

Comment:
I haven't tried this myself but the docs indicate this should be `scope.transaction` not `scope.transaction.name`. The `if scope.transaction` part also seems unnecessary.

See https://docs.sentry.io/platforms/python/guides/logging/enriching-events/transaction-name/